common/error.h

```diff
@@ -23,19 +23,48 @@ struct Success {};
 class [[nodiscard]] Error {
  public:
   // Represents an error state.
-  explicit Error(llvm::Twine message) : message_(message.str()) {
+  explicit Error(llvm::Twine prefix, llvm::Twine location, llvm::Twine message)
```

Comment:
```suggestion
  Error(llvm::Twine prefix, llvm::Twine location, llvm::Twine message)
```

Reply:
I think this `explicit` is beneficial. I want to prevent copy-list-initialization from three strings, [for example](https://godbolt.org/z/hnab6cndh):

```
ErrorOr<Something> DoSomething()  {
  return {{a, b, c}}; // should never create an Error even if these are all strings
}
```

In general, we want an `Error` to only be constructed by explicitly mentioning its type name, so all of its constructors should be `explicit`.